### PR TITLE
ICU-20605 travis: make dist.

### PR DESCRIPTION
**Pull request by srl295**
[ICU-20605]

- run 'make dist' after the linux clang build
~- upgrade ubuntu xenial (16 LTS) to bionic (18 LTS) [(schedule)](https://wiki.ubuntu.com/Releases)~

HT: https://github.com/unicode-org/icu/pull/1048#discussion_r392544485

[ICU-20605]: https://unicode-org.atlassian.net/browse/ICU-20605

### CODE REVIEW

**Comment by srl295** [comment]
This should maybe be two PRs. One for make dist, and one for bumping the version.

Or should we keep trying ubuntu xenial?

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ matrix:
     #    TODO: publish the latest built docs from master on GitHub pages.
     #          See https://gist.github.com/vidavidorra/548ffbcdae99d752da02 for ideas.
     - name:     "c: docs"
-      dist:     xenial
+      dist:     bionic
```
Comment (jefgen) [inline .travis.yml]: oof, should likely update (or add) to azure pipelines bionic as well

File: .travis.yml [hunk 2]
```diff
@@ -87,6 +87,7 @@ matrix:
         - make -j2 check
         - ( cd test/depstest && python3 depstest.py ../../../source/ )
         - ( cd .. && source/test/hdrtst/testtagsguards.sh )
+        - make dist # not sure if this is -j safe everywhere.
```
Comment (jefgen) [inline .travis.yml]: does travis let you download artifacts?

Comment (srl295) [reply .travis.yml]: not sure. We could push them somewhere (like a github release) though.

**Comment by jefgen** [comment]
I'd vote for a separate ticket to bump the version. We should probably bump the azure pipelines one too at the same time. (or add an extra build line)

File: .travis.yml (hunk 2 above)
Comment (jefgen) [reply .travis.yml]: fwiw, you can publish artifacts from the pipelines.
ex: https://github.com/unicode-org/icu/blob/master/.ci-builds/.azure-pipelines.yml#L108
(This is how the windows binaries are produced.)
It should work for *nix as well, so we might be able to have a sort of rolling/evergreen make dist available.

Comment (jefgen) [reply .travis.yml]: It looks like Travis can potentially directly deploy to a GitHub release:
https://docs.travis-ci.com/user/build-stages/deploy-github-releases/
which is definitely nicer.
AFAIK, the pipelines just give you an arbitrary file you can download. (So you'd have to copy/paste to a release).

Comment (srl295) [reply .travis.yml]: right. But I'm not sure what release it would go to.

**Comment by sffc** [comment]
Your new check is already failing

```
make[3]: *** No rule to make target 'icu-docs.zip'.  Stop.
```

**Pull request merged**